Classes/Service/TOTPService.php

```diff
@@ -38,14 +44,14 @@ public static function checkIfOtpIsValid(string $secret, string $submittedOtp):
     public function generateQRCodeForTokenAndAccount(TOTP $otp, Account $account): string
     {
         $secret = $otp->getSecret();
-
         $currentDomain = $this->domainRepository->findOneByActiveRequest();
         $currentSite = $currentDomain !== null ? $currentDomain->getSite() : $this->siteRepository->findDefault();
         $currentSiteName = $currentSite->getName();
         $urlEncodedSiteName = urlencode($currentSiteName);
-
         $userIdentifier = $account->getAccountIdentifier();
-        $oauthData = "otpauth://totp/$userIdentifier?secret=$secret&period=30&issuer=$urlEncodedSiteName";
+        // If the issuerName is set in the configuration, use that. Else fall back to the default.
+        $issuer = $this->issuerName != '' ? urlencode($this->issuerName) : $urlEncodedSiteName;
```

Comment:
To prevent a breaking change, please make this setting truly optional with

```php
$issuer = !empty($this->issuerName) ? urlencode...
```

and set the type of the injected property to `string | null`.

Reply:
@JamesAlias thanks for taking a look. Added the changes in the last 2 commits.